apps/src/music/views/MusicView.jsx

```diff
@@ -100,9 +102,16 @@ class UnconnectedMusicView extends React.Component {
     // TODO: the 'beforeunload' callback is advised against as it is not guaranteed to fire on mobile browsers. However,
     // we need a way of reporting analytics when the user navigates away from the page. Check with Amplitude for the
     // correct approach.
-    window.addEventListener('beforeunload', () =>
-      this.analyticsReporter.endSession()
-    );
+    window.addEventListener('beforeunload', event => {
+      this.analyticsReporter.endSession();
+      // force a save before the page unloads, if there are unsaved changes.
```

Comment:
general nit: all comments should begin with an upper-case letter, end with a period, etc.